src/group_theory/specific_groups/alternating.lean

```diff
@@ -0,0 +1,128 @@
+consisting of the even permutations.
+
+## Main definitions
+
+* `alternating_group α` is the alternating group on `α`, defined as a `subgroup (perm α)`.
+
+## Main results
+* `two_mul_card_alternating_group` shows that the alternating group is half as large as
+  the permutation group it is a subgroup of.
+
+* `closure_three_cycles_eq_alternating` shows that the alternating group is
+  generated by three-cycles.
+
+## Tags
+alternating group permutation
+
+
+## TODO
+* Show that `alternating_group α` is simple if and only if `fintype.card α ≠ 4`.
+
+-/
+
+open equiv equiv.perm subgroup fintype
+variables (α : Type*) [fintype α] [decidable_eq α]
+
+/-- The alternating group on a finite type, realized as a subgroup of `equiv.perm`.
+  For $A_n$, use `alternating_group (fin n)`. -/
+@[derive fintype] def alternating_group : subgroup (perm α) :=
+sign.ker
+
+instance [subsingleton α] : unique (alternating_group α) :=
+⟨⟨1⟩, λ ⟨p, hp⟩, subtype.eq (subsingleton.elim p _)⟩
+
+variables {α}
+
+lemma alternating_group_eq_sign_ker : alternating_group α = sign.ker := rfl
+
+namespace equiv.perm
+
+@[simp]
+lemma mem_alternating_group {f : perm α} :
+  f ∈ alternating_group α ↔ sign f = 1 :=
+sign.mem_ker
+
+lemma prod_list_swap_mem_alternating_group_iff_even_length {l : list (perm α)}
+  (hl : ∀ g ∈ l, is_swap g) :
+  l.prod ∈ alternating_group α ↔ even l.length :=
+begin
+  rw [mem_alternating_group, sign_prod_list_swap hl, ← units.coe_eq_one, units.coe_pow,
+    units.coe_neg_one, nat.neg_one_pow_eq_one_iff_even],
+  dec_trivial
+end
+
+end equiv.perm
+
+lemma two_mul_card_alternating_group [nontrivial α] :
+  2 * card (alternating_group α) = card (perm α) :=
+begin
+  let := (quotient_group.quotient_ker_equiv_of_surjective _ (sign_surjective α)).to_equiv,
+  rw [←fintype.card_units_int, ←fintype.card_congr this],
+  exact (subgroup.card_eq_card_quotient_mul_card_subgroup _).symm,
+end
+
+instance alternating_group_normal : (alternating_group α).normal := sign.normal_ker
+
+namespace equiv.perm
+
+lemma swap_mul_swap_same_mem_closure_three_cycles {a b c : α} (ab : a ≠ b) (ac : a ≠ c) :
+  (swap a b * swap a c) ∈ closure {σ : perm α | is_three_cycle σ } :=
+begin
+  by_cases bc : b = c,
+  { subst bc,
+    simp [one_mem] },
+  exact subset_closure (is_three_cycle_swap_mul_swap_same ab ac bc)
+end
+
+lemma is_swap.mul_mem_closure_three_cycles {σ τ : perm α}
+  (hσ : is_swap σ) (hτ : is_swap τ) :
+  σ * τ ∈ closure {σ : perm α | is_three_cycle σ } :=
+begin
+  obtain ⟨a, b, ab, rfl⟩ := hσ,
+  obtain ⟨c, d, cd, rfl⟩ := hτ,
+  by_cases ac : a = c,
+  { subst ac,
+    exact swap_mul_swap_same_mem_closure_three_cycles ab cd },
+  have h' : swap a b * swap c d = swap a b * swap a c * (swap c a * swap c d),
+  { simp [swap_comm c a, mul_assoc] },
+  rw h',
+  exact mul_mem _ (swap_mul_swap_same_mem_closure_three_cycles ab ac)
+    (swap_mul_swap_same_mem_closure_three_cycles (ne.symm ac) cd),
+end
```

Comment:
These statements feel like they belong in `cycle_type`, since they don't mention `alternating_group`